docs/se/oci/02_object-storage.adoc

```diff
@@ -0,0 +1,203 @@
+        </dependency>
+----
+
+
+== Setting up the Object Storage
+
+
+In order to use the OCI Object Storage integration, the following setup should be made:
+
+[source,java]
+----
+Config ociConfig = config.get("oci");
+
+OciObjectStorageRx ociObjectStorage = OciObjectStorageRx.create(ociConfig);
+----
+
+Current configuration requires `~/.oci/config` to be available in the home folder. This configuration file can be downloaded from OCI.
+
+`Routing` should be added to the `WebServer`, in our case pointing to `/file`:
+
+[source,java]
+----
+String bucketName = ociConfig.get("objectstorage").get("bucket").asString().get();
+
+        WebServer.builder()
+                .config(config.get("server"))
+                .routing(Routing.builder()
+                                 .register("/files", new ObjectStorageService(ociObjectStorage, bucketName)))
+                .build()
+                .start()
+                .await()
+----
+
+Additionally, in `application.yaml` OCI properties should be specified:
+
+[source,yaml]
+----
+oci:
+    properties:
+      compartment-ocid: "ocid<1>tenancy.oc<1>.<..>"
+      objectstorage-namespace: "<...>"
+      objectstorage-bucket: "<...>"
+
+----
+
+The exact values are available in OCI object storage and bucket properties.
+
+image::oci/ocibucket.png[OCI Bucket, align="center"]
+
+== Using the Object Storage
+
+In the Service we must specify the mapping for CRUD operations with the files and their handlers:
+
+[source,java]
+----
+@Override
+public void update(Routing.Rules rules) {
+    rules.get("/file/{file-name}", this::download)
+            .post("/file/{file-name}", this::upload)
+            .delete("/file/{file-name}", this::delete)
+            .get("/rename/{old-name}/{new-name}", this::rename);
+}
+----
+
+To upload a file via `PUT` method to OCI Object Storage:
+
+[source,java]
+----
+private void upload(ServerRequest req, ServerResponse res) {
+    OptionalLong contentLength = req.headers().contentLength();
+    if (contentLength.isEmpty()) {
+        req.content().forEach(DataChunk::release);
+        res.status(Http.Status.BAD_REQUEST_400).send("Content length must be defined");
+        return;
+    }
+
+    String objectName = req.path().param("file-name");
+
+    PutObject.Request request = PutObject.Request.builder() <1>
+            .objectName(objectName)
+            .bucket(bucketName)
+            .contentLength(contentLength.getAsLong());
+
+    req.headers().contentType().ifPresent(request::requestMediaType); <2>
+
+    objectStorage.putObject(request,
+                            req.content())
+            .forSingle(response -> res.send(response.requestId())) <3>
+            .exceptionally(res::send);
+}
+----
+
+<1> Create the `Request` using `PutObject.Request.builder()`
+<2> Define `MediaType`
+<3> Execute the request to OCI in asynchronous way and put the result in `response` object
+
+To download the file via `GET` method from OCI Object Storage:
```

Comment:
To download a file from OCI Object Storage using the `GET` method:

Reply:
fixed